web/common.inc.sh

```diff
@@ -85,17 +85,36 @@ function prepare() {
 # ```
 function test-headless() {
   TEST_FOLDER=$1
-  TEST_BASE=${2:-${KEYMAN_ROOT}/web/src/test/auto/headless/}
+  TEST_EXTENSIONS=${2:-}
+  TEST_BASE="${KEYMAN_ROOT}/web/src/test/auto/headless/"
 
   TEST_OPTS=
+  TEST_CD_REQD=false
   if builder_has_option --ci; then
     TEST_OPTS="--reporter mocha-teamcity-reporter"
   fi
+  if [[ -n "$TEST_EXTENSIONS" ]]; then
+    TEST_OPTS=" --extension $TEST_EXTENSIONS"
+    TEST_CD_REQD=true
+  fi
+
+  if [ $TEST_CD_REQD ]; then
+    # The mocha config needed to live-compile TS-based tests only applies
+    # if the command is started within the appropriate subfolder.
+    pushd "${TEST_BASE}" > /dev/null
+    TEST_BASE=
+  else
+    TEST_BASE="${TEST_BASE}/"
```

Comment:
It probably doesn't hurt, but we don't need this `else` block since `TEST_BASE` already has the trailing slash (line 89).